Sources/AblyChat/DefaultMessages.swift

```diff
@@ -0,0 +1,237 @@
+        let messageSubscription = MessageSubscription(
+            bufferingPolicy: bufferingPolicy
+        ) { [weak self] queryOptions in
+            guard let self else { throw MessagesError.noReferenceToSelf }
+            return try await getBeforeSubscriptionStart(uuid, params: queryOptions)
+        }
+
+        subscriptionPoints[uuid] = .init(subscription: messageSubscription, timeserial: timeserial)
+
+        // (CHA-M4c) When a realtime message with name set to message.created is received, it is translated into a message event, which contains a type field with the event type as well as a message field containing the Message Struct. This event is then broadcast to all subscribers.
+        // (CHA-M4d) If a realtime message with an unknown name is received, the SDK shall silently discard the message, though it may log at DEBUG or TRACE level.
+        // (CHA-M5d) Incoming realtime events that are malformed (unknown field should be ignored) shall not be emitted to subscribers.
+        channel.subscribe(MessageEvent.created.rawValue) { message in
+            Task {
+                guard let data = message.data as? [String: Any],
+                      let text = data["text"] as? String
+                else {
+                    return
+                }
+
+                guard let timeserial = try message.extras?.toJSON()["timeserial"] as? String else {
+                    return
+                }
+
+                let message = Message(
+                    timeserial: timeserial,
+                    clientID: message.clientId,
+                    roomID: self.roomID,
+                    text: text,
+                    createdAt: message.timestamp,
+                    metadata: .init(),
+                    headers: .init()
+                )
+
+                messageSubscription.emit(message)
+            }
+        }
+
+        return messageSubscription
+    }
+
+    // (CHA-M6a) A method must be exposed that accepts the standard Ably REST API query parameters. It shall call the “REST API”#rest-fetching-messages and return a PaginatedResult containing messages, which can then be paginated through.
+    public func get(options: QueryOptions) async throws -> any PaginatedResult<Message> {
+        try await chatAPI.getMessages(roomId: roomID, params: options)
+    }
+
+    public func send(params: SendMessageParams) async throws -> Message {
+        try await chatAPI.sendMessage(roomId: roomID, params: params)
+    }
+
+    // TODO: (CHA-M7) Users may subscribe to discontinuity events to know when there’s been a break in messages that they need to resolve. Their listener will be called when a discontinuity event is triggered from the room lifecycle. - https://github.com/ably-labs/ably-chat-swift/issues/47
+    public nonisolated func subscribeToDiscontinuities() -> Subscription<ARTErrorInfo> {
+        fatalError("not implemented")
+    }
+
+    public nonisolated func discontinuityDetected(reason: ARTErrorInfo?) {
```

Comment:
Is this needed?

Reply:
not anymore...  older protocol conformance - will remove